src/libraries/System.Security.Cryptography.SP800108/src/System/Security/Cryptography/NetStandardShims.cs

```diff
@@ -0,0 +1,34 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Text;
+using System.Runtime.CompilerServices;
+
+namespace System.Security.Cryptography
+{
+    internal static class NetStandardShims
+    {
+        internal static unsafe int GetBytes(this Encoding encoding, ReadOnlySpan<char> str, Span<byte> destination)
```

Comment:
I wonder if this should be moved to a common helper file and shared across projects. System.Text.Json has similar code, as shown below. (And both this file and the JSON copy have subtle bugs when handling zero-length source \& destination, which doesn't matter _here_ but we should probably address nonetheless if we're going to get in the habit of copying this code.)

https://github.com/dotnet/runtime/blob/c6d6b19da5427ae78eab241d2af596a3320a5e5b/src/libraries/System.Text.Json/src/System/Text/Json/Reader/JsonReaderHelper.Unescaping.cs#L247-L258

Reply:
The subtle bug being... what? When a BOM needs to get written?

Reply:
> The subtle bug being... what? When a BOM needs to get written?

The `GetChars` method can't handle null inputs or outputs, and _in the general case_ there's no guarantee that a zero-length input produces a zero-length output. Doesn't matter for UTF-8, but I'm seeing this code duplicated in a few places and we may as well just write it once and get it correct from the beginning.

Reply:
That example is not exactly the same. That one is doing `GetChars`. We are doing `GetBytes`. So we can't really consolidate these two specific examples. I can extract out helpers if you want, but in this particular case it won't result in de-duping anything.